distributed/shuffle/tests/test_rechunk.py

```diff
@@ -1350,4 +1351,39 @@ async def test_partial_rechunk_taskgroups(c, s):
         ),
         timeout=5,
     )
-    assert len(s.task_groups) < 7
+    assert len(s.task_groups) < 8
+
+
+@pytest.mark.parametrize(
+    ["old", "new", "expected"],
+    [
+        [((2, 2),), ((2, 2),), ((2, 2),)],
+        [((2, 2),), ((4,),), ((2, 2),)],
+        [((2, 2),), ((1, 1, 1, 1),), ((2, 2),)],
+        [((2, 2, 2),), ((1, 2, 2, 1),), ((1, 1, 1, 1, 2),)],
```

Comment:
Why is the last 2 in expected a good idea?

Reply:
Is it because the 1 in new at the end means that the input chunk only creates 2 output chunks?

Reply:
The basic idea is that culling doesn't benefit from splitting the last two output chunks and there's "all-to-all" communication.

Reply:
sounds good

Reply:
I've thought about this a little more, and you're right! There was a small bug here. I've adjusted the algorithm accordingly.